src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -802,6 +802,83 @@ static double validateAndParseReal(String columnName, Object input, long insertR
         columnName, input.getClass(), "REAL", new String[] {"Number", "String"}, insertRowIndex);
   }
 
+  /**
+   * Validates and parses input to a specific Iceberg column number type including INT, LONG, FLOAT,
+   * and DOUBLE. Allowed Java types:
+   *
+   * <ul>
+   *   <li>Number
+   *   <li>String
+   * </ul>
+   *
+   * @param columnName Column name, used in validation error messages
+   * @param input Object to validate and parse
+   * @param insertRowIndex Row index for error reporting
+   * @param targetType The type to which the input should be parsed ("INT", "LONG", "FLOAT",
+   *     "DOUBLE")
+   * @return Parsed number as Number to be casted by the caller
+   */
+  static <T extends Number> T validateAndParseIcebergNumber(
+      String columnName, Object input, long insertRowIndex, Class<T> targetType) {
+    if (input instanceof Number) {
+      if (targetType.equals(Integer.class)) {
+        return targetType.cast(((Number) input).intValue());
```

Comment:
Can you check what's the difference between type.class.cast() and Number.intValue / Number.longValue / Number.floatValue / Number.doubleValue ?

Reply:
Tye `type.class.cast` is used for the return type check and it's no longer needed after we split it into 4 different methods. The `Number.xValue` do narrow casting if the source type cannot fit into target type.